src/app/routes/admin-forms.server.routes.js

```diff
@@ -92,7 +93,41 @@ module.exports = function (app) {
   app
     .route('/adminform')
     .get(withUserAuthentication, AdminFormController.handleListDashboardForms)
-    .post(withUserAuthentication, adminForms.create)
+    .post(
+      withUserAuthentication,
+      celebrate({
+        [Segments.BODY]: {
+          form: Joi.object()
+            .keys({
+              // Require valid responsesMode field.
+              responseMode: Joi.string()
+                .valid(...Object.values(ResponseMode))
+                .required(),
+              // Require title field.
+              title: Joi.string().min(4).max(200).required(),
+              // Require emails string (for backwards compatibility) or string
+              // array if form to be created in Email mode.
+              emails: Joi.alternatives()
+                .try(Joi.array().items(Joi.string()), Joi.string())
+                .when('responseMode', {
+                  is: ResponseMode.Email,
+                  then: Joi.required(),
+                }),
+              // Require publicKey field if form to be created in Storage mode.
+              publicKey: Joi.string()
+                .allow('')
+                .when('responseMode', {
+                  is: ResponseMode.Encrypt,
+                  then: Joi.string().required().disallow(''),
+                }),
+            })
+            .required()
+            // Allow other form schema keys to be passed for form creation.
+            .unknown(true),
```

Comment:
is this necessary?

Reply:
Yes, because any form schema key should work, and any invalid keys should be caught by the mongoose schema validator.  We can use something like joigoose to validate, but it's P3. Created issue https://github.com/opengovsg/FormSG/issues/814